utils/post.js

```diff
@@ -341,7 +341,17 @@ const modifyNewItemAnonymity = (newItem, isBlurredPost = false) => {
   return newItem;
 };
 
+const isUserFollowAuthor = async (newItem, followingUsers) => {
+  if (newItem?.actor?.id) {
+    newItem.is_following_target = followingUsers.includes(newItem?.actor?.id);
+  }
+
+  return newItem;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `isUserFollowAuthor` checks if the user is following the author of a post. However, it doesn't handle the case when `newItem.actor` is undefined or null. This could lead to an error if `newItem.actor` is not defined. Consider adding error handling for this scenario.
```diff
const isUserFollowAuthor = async (newItem, followingUsers) => {
  if (newItem?.actor?.id) {
    newItem.is_following_target = followingUsers.includes(newItem?.actor?.id);
  } else {
    throw new Error('Actor is not defined in the item');
  }
  return newItem;
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->